src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1003,12 +1003,12 @@ bool IsDelegateAssignable(MethodDesc ftn, TypeDesc delegateType)
 
             for (int i = 0; i < ftnSignature.Length; i++)
             {
-                if (!IsAssignable(ftnSignature[i], delegateSignature[i]))
+                if (!IsAssignable(delegateSignature[i], ftnSignature[i]))
```

Comment:
Technically the types should be "Compatible". In the spec "Compatible" is more strict than "Assignable", since "Assignable" allows some reinterpretations. I.E. an `int32` is Assignable to `int8` - i.e. can be assigned with implicit value truncation.  
However `int32` and `int8` are not "Compatible" and therefore  ` int M1() ` cannot be used as `Func<byte>`

I am not sure what `IsAssignable` implements. It could as well implement "IsCompatible", then everything is correct.  
Perhaps it is worth adding a test while touching this?

Reply:
I am sorry - I have not noticed that you have commented on this before merging. I agree it would be nice to add test for this case.

Reply:
Filed follow-up issue.

Reply:
I would not block merging on this. Just a good thing to have.